Constructor for `AlgGrp` assumes that the neutral element comes first

The constructor for `AlgGrp` assumes that `enumerate(G)` returns the neutral element of `G` as the first entry.
Here is an example with OSCAR:
```
julia> K, a = CyclotomicField(3);

julia> r = matrix(K, 2, 2, [ 1, 0, 0, a ]);

julia> s = matrix(K, 2, 2, [ 1//3*a - 1//3*a^2, 2//3*a + 1//3*a^2, 4//3*a + 2//3*a^2, -1//3*a - 2//3*a^2 ]);

julia> G = matrix_group(r, s);

julia> group_algebra(K, G)
ERROR: AssertionError: all((A.mult_table[1, i] == i for i = 1:dim(A)))
Stacktrace:
 [1] (::Hecke.var"#93#96"{typeof(*), AnticNumberField, MatrixGroup{nf_elem, AbstractAlgebra.Generic.MatSpaceElem{nf_elem}}})()
   @ Hecke ~/.julia/dev/Hecke/src/AlgAss/Types.jl:190
 [2] get!
   @ ./iddict.jl:178 [inlined]
 [3] get_cached!
   @ ~/.julia/packages/AbstractAlgebra/t6A7h/src/AbstractAlgebra.jl:340 [inlined]
 [4] AlgGrp
   @ ~/.julia/dev/Hecke/src/AlgAss/Types.jl:158 [inlined]
 [5] #group_algebra#3584
   @ ~/.julia/dev/Hecke/src/AlgAss/AlgGrp.jl:68 [inlined]
 [6] group_algebra(K::AnticNumberField, G::MatrixGroup{nf_elem, AbstractAlgebra.Generic.MatSpaceElem{nf_elem}})
   @ Hecke ~/.julia/dev/Hecke/src/AlgAss/AlgGrp.jl:68
 [7] top-level scope
   @ REPL[8]:1
```
I tried to fix this with https://github.com/joschmitt/Hecke.jl/commit/ec929cc581ff7f862f99810e6a6f968d7c2dfde2 , but then the Hecke tests don't pass because `one(G)` is missing for  `GrpAbFinGen` and also the groups one gets with `small_group`.

--- a/src/AlgAss/Types.jl
+++ b/src/AlgAss/Types.jl
@@ -133,7 +133,7 @@ end
   base_ring::Ring
   group::S
   group_to_base::Dict{R, Int}
-  base_to_group::Dict{Int, R}
+  base_to_group::Vector{R}
   one::Vector{T}
   basis#::Vector{AlgAssElem{T, AlgAss{T}}
   gens#::Vector{AlgAssElem{T, AlgAss{T}} # "Small" number of algebra generators
@@ -164,12 +164,20 @@ end
       A.group = G
       d = Int(order(G))
       A.group_to_base = Dict{elem_type(G), Int}()
-      A.base_to_group = Dict{Int, elem_type(G)}()
+      A.base_to_group = Vector{elem_type(G)}(undef, d)
       A.mult_table = zeros(Int, d, d)
 
-      for (i, g) in enumerate(G)
+      A.group_to_base[deepcopy(one(G))] = 1
+      A.base_to_group[1] = deepcopy(one(G))
+
+      i = 2
+      for g in collect(G)
+        if isone(g)
+          continue
+        end
         A.group_to_base[deepcopy(g)] = i
         A.base_to_group[i] = deepcopy(g)
+        i += 1
       end
 
       v = Vector{elem_type(K)}(undef, d)
@@ -180,9 +188,9 @@ end
 
       A.one = v
 
-      for (i, g) in enumerate(G)
-        for (j, h) in enumerate(G)
-          l = op(g, h)
+      for i = 1:d
+        for j = 1:d
+          l = op(A.base_to_group[i], A.base_to_group[j])
           A.mult_table[i, j] = A.group_to_base[l]
         end
       end

--- a/src/AlgAss/Types.jl
+++ b/src/AlgAss/Types.jl
@@ -167,12 +167,11 @@ end
       A.base_to_group = Vector{elem_type(G)}(undef, d)
       A.mult_table = zeros(Int, d, d)
 
-      A.group_to_base[deepcopy(one(G))] = 1
-      A.base_to_group[1] = deepcopy(one(G))
-
       i = 2
       for g in collect(G)
         if isone(g)
+          A.group_to_base[deepcopy(g)] = 1
+          A.base_to_group[1] = deepcopy(g)
           continue
         end
         A.group_to_base[deepcopy(g)] = i

